src/runtime_src/hip/core/context.cpp

```diff
@@ -45,5 +46,21 @@ get_current_context()
   auto ctx_hdl = hip_device_primary_ctx_retain(tls_objs.dev_hdl);
   return context_cache.get(ctx_hdl);
 }
+
+std::shared_ptr<stream>
+context::
+get_null_stream()
+{
+  std::shared_ptr<stream> s = null_stream.lock();
+  if (s)
+    return s;
+
+  // create null stream
+  auto null_s = std::make_shared<stream>(shared_from_this(), hipStreamDefault, true);
+  // assign weak ptr for future use
+  null_stream = null_s;
+  insert_in_map(stream_cache, std::move(null_s));
+  return null_stream.lock(); // inserted in map, so we can get shared_ptr from weak_ptr
```

Comment:
The comment indicates that we have already created shared_ptr and inserted in map so locking weak ptr ensures we get shared ptr and not a null value. I will rephrase the comment.
please read the doc - https://docs.nvidia.com/cuda/cuda-driver-api/stream-sync-behavior.html#stream-sync-behavior
to understand more about null stream and its relation with context.

Reply:
Thanks for the link.
OK, by "null" stream you mean default stream.
So, why not renaming to something like `default_stream` to be in sync with HIP and CUDA naming instead of using a confusing one?

Reply:
There are two kinds of default streams per thread default stream and legacy default stream which is called null stream. In the doc also they use null stream for legacy default stream, so I feel null stream would represent it properly

Reply:
Thanks for the clarification.
But then, why not call it `Legacy default stream` since it is what is used in https://docs.nvidia.com/cuda/cuda-driver-api/stream-sync-behavior.html#stream-sync-behavior ?

Reply:
Yes you are right but our HIP documentation and code uses the name null stream for legacy default stream, so using this name.

Reply:
Oh, I see, this follows the *HIP* naming, not the *CUDA* naming.
I thought the goal of HIP was to be a clone of CUDA. So perhaps the HIP team should fix this at the first place? @saadrahim @b-sumner
So perhaps we should follow the CUDA naming, independently of HIP, to avoid too much technical debt?